spark/src/main/scala/org/apache/spark/sql/delta/skipping/clustering/ClusteredTableUtils.scala

```diff
@@ -41,3 +60,160 @@ trait ClusteredTableUtilsBase extends DeltaLogging {
 }
 
 object ClusteredTableUtils extends ClusteredTableUtilsBase
+
+/**
+ * A container for clustering information. Copied from OSS Spark.
+ *
+ * This class will be removed when we integrate with OSS Spark's CLUSTER BY implementation.
+ * @see https://github.com/apache/spark/pull/42577
+ *
+ * @param columnNames the names of the columns used for clustering.
+ */
+case class ClusterBySpec(columnNames: Seq[NamedReference]) {
```

Comment:
Maybe name this to `TempClusterBySpec` to make it obvious?

Reply:
Moved to its own package `clustering.temp` and file `ClusterBySpec.scala`.